scripts/provision_keys.go

```diff
@@ -79,4 +79,116 @@ func revokeAuthorizedKey(requestID, authorizedKeysPath string, logger *logrus.Lo
 		Success: true,
 		Message: fmt.Sprintf("SSH public key removed from %s successfully", authorizedKeysPath),
 	}
-}
\ No newline at end of file
+}
+
+// ProvisionCAKeys provisions CA public keys with cert-authority and principals parameters
+func ProvisionCAKeys(req ProvisioningRequest, logger *logrus.Logger) ProvisioningResult {
+	logger.WithFields(logrus.Fields{
+		"username":   req.UserName,
+		"action":     req.Action,
+		"request_id": req.RequestID,
+		"has_ca_key": req.CAPublicKey != "" && req.CAPublicKey != "N/A",
+	}).Info("🔐 Provisioning CA keys")
+
+	if (req.CAPublicKey == "" || req.CAPublicKey == "N/A") && req.Action == "grant" {
+		return ProvisioningResult{
+			Success: true,
+			Message: "No CA public key provided, skipping CA keys provisioning",
+		}
+	}
+
+	userInfo, err := user.Lookup(req.UserName)
+	if err != nil {
+		return ProvisioningResult{
+			Success: false,
+			Error:   fmt.Sprintf("user %s not found: %v", req.UserName, err),
+		}
+	}
+
+	authorizedKeysPath := filepath.Join(userInfo.HomeDir, ".ssh", "authorized_keys")
+
+	switch req.Action {
+	case "grant":
+		return grantCAKey(req.CAPublicKey, req.RequestID, authorizedKeysPath, req.UserName, logger)
+	case "revoke":
+		return revokeCAKey(req.RequestID, authorizedKeysPath, logger)
+	default:
+		return ProvisioningResult{
+			Success: false,
+			Error:   "invalid action: must be 'grant' or 'revoke'",
+		}
+	}
+}
+
+func grantCAKey(caPublicKey, requestID, authorizedKeysPath, username string, logger *logrus.Logger) ProvisioningResult {
+	logger.WithFields(logrus.Fields{
+		"path":       authorizedKeysPath,
+		"username":   username,
+		"request_id": requestID,
+	}).Debug("Granting CA key access")
+
+	// Format CA key with cert-authority and principals parameters
+	caKeyEntry := fmt.Sprintf("cert-authority,principals=\"%s\" %s", username, caPublicKey)
+
+	result := ensureContentInFile(caKeyEntry, requestID, authorizedKeysPath, "600", username, logger)
+	if !result.Success {
+		return result
+	}
+
+	return ProvisioningResult{
+		Success: true,
+		Message: fmt.Sprintf("CA public key added to %s successfully with cert-authority,principals=\"%s\"", authorizedKeysPath, username),
```

Comment:
nit: use the caKeyEntry variable
```suggestion
		Message: fmt.Sprintf("CA public key added to %s successfully with %s", authorizedKeysPath, caKeyEntry),
```